scripts/pylib/twister/twisterlib.py

```diff
@@ -3553,21 +3552,17 @@ def execute(self, pipeline, done, results):
             for p in processes:
                 p.terminate()
 
-        # FIXME: This needs to move out.
+        return results
+
+    def gather_footprints(self, instance):
         if self.enable_size_report and not self.cmake_only:
-            # Parallelize size calculation
-            executor = concurrent.futures.ThreadPoolExecutor(self.jobs)
-            futures = [executor.submit(self.calc_one_elf_size, instance)
-                       for instance in self.instances.values()]
-            concurrent.futures.wait(futures)
+            self.calc_one_elf_size(instance)
         else:
-            for instance in self.instances.values():
-                instance.metrics["ram_size"] = 0
-                instance.metrics["rom_size"] = 0
-                instance.metrics["handler_time"] = instance.handler.duration if instance.handler else 0
-                instance.metrics["unrecognized"] = []
+            instance.metrics["ram_size"] = 0
+            instance.metrics["rom_size"] = 0
+            instance.metrics["handler_time"] = instance.handler.duration if instance.handler else 0
```

Comment:
if in build stage, the handler_time is always 0, it will be calculated and set in run stage later, so it seems that we don't need to set it in this gather_footprints.

Reply:
thanks, done